addons/binding/org.openhab.binding.mihome.test/src/test/groovy/org/openhab/binding/mihome/test/JsonSubdevice.groovy

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mihome.test;
+
+/**
+ * JSON representation of a subdevice used in the communication with the server.
+ *
+ * @author Svilen Valkanov
+ */
+public class JsonSubdevice extends JsonDevice {
+
+    public static final String DEFAULT_LABEL = "New Subdevice"
+    // ID of the gateway that the subdevice is paired to
+    int device_id
+    //For open and motion sensors
+    Integer sensor_state
+
+    //Only for energy monitor
+    double voltage
+    int real_power
+    int today_wh
+
+    JsonSubdevice(int id,int gatewayID,String type) {
```

Comment:
I don't think it is necessary, this is enough for test purposes.
Actually in the MiH|ome API, all subdevices have the same representation.